src/main/java/com/intel/genomicsdb/GenomicsDBImporter.java

```diff
@@ -731,23 +649,42 @@ static File printLoaderJSONFile(
    *
    * Assume one sample per input GVCF file
    *
-   * @param  variants  Variant Readers objects of the input GVCF files
+   * @param  sampleToReaderMap  Variant Readers objects of the input GVCF files
    * @param useSamplesInOrderProvided  If True, do not sort the samples,
    *                                   use the order they appear in
    * @param lbRowIdx Smallest row idx which should be imported by this object
    * @return  Mappings of callset (sample) names to TileDB rows
    */
   static GenomicsDBCallsetsMapProto.CallsetMappingPB generateSortedCallSetMap(
-      final Map<String, FeatureReader<VariantContext>> variants,
+      final Map<String, FeatureReader<VariantContext>> sampleToReaderMap,
       boolean useSamplesInOrderProvided,
       final long lbRowIdx)
   {
-    List<String> sampleNames = new ArrayList<>(variants.size());
-    for (Map.Entry<String, FeatureReader<VariantContext>> v : variants.entrySet()) {
-      VCFHeader h = (VCFHeader) v.getValue().getHeader();
-      sampleNames.add(h.getSampleNamesInOrder().get(0));
+    List<String> sampleNames = new ArrayList<>(sampleToReaderMap.size());
+
+    for (Map.Entry<String, FeatureReader<VariantContext>> mapObject :
+        sampleToReaderMap.entrySet()) {
+
+      String sampleName = mapObject.getKey();
+      FeatureReader<VariantContext> featureReader = mapObject.getValue();
+      VCFHeader header = (VCFHeader) featureReader.getHeader();
+      List<String> sampleNamesInHeader = header.getSampleNamesInOrder();
+      if (sampleNamesInHeader.size() > 1) {
+        String message = "Multiple samples ";
+        for (String name : sampleNamesInHeader) {
+          message.concat(name + " ");
+        }
+        message.concat(" appear in header for " + sampleName);
+        throw new IllegalArgumentException(message);
+      } else {
+        if (!sampleName.equals(sampleNamesInHeader.get(0))) {
+          System.err.println("Sample does not match with header for " + sampleName);
```

Comment:
This message should include the sample name that was passed in AND the sample name from the header.